public/app/features/serviceaccounts/state/reducers.ts

```diff
@@ -31,19 +29,53 @@ export const serviceAccountProfileSlice = createSlice({
   },
 });
 
+export const serviceAccountProfileReducer = serviceAccountProfileSlice.reducer;
+export const { serviceAccountLoaded, serviceAccountTokensLoaded } = serviceAccountProfileSlice.actions;
+
+// serviceAccountsListPage
+export const initialStateList: ServiceAccountsState = {
+  serviceAccounts: [] as ServiceAccountDTO[],
+  isLoading: true,
+  builtInRoles: {},
+  roleOptions: [],
+  serviceAccountToRemove: null,
+  query: '',
+  page: 0,
+  perPage: 50,
+  totalPages: 1,
+  showPaging: false,
+  filters: [{ name: 'Expired', value: true }],
+};
+
+interface ServiceAccountsFetched {
+  serviceAccounts: ServiceAccountDTO[];
+  perPage: number;
+  page: number;
+  totalCount: number;
+}
+
 const serviceAccountsSlice = createSlice({
   name: 'serviceaccounts',
-  initialState,
+  initialState: initialStateList,
   reducers: {
-    serviceAccountsLoaded: (state, action: PayloadAction<ServiceAccountDTO[]>): ServiceAccountsState => {
-      return { ...state, isLoading: false, serviceAccounts: action.payload };
+    serviceAccountsFetched: (state, action: PayloadAction<ServiceAccountsFetched>): ServiceAccountsState => {
+      const { totalCount, perPage, ...rest } = action.payload;
+      const totalPages = Math.ceil(totalCount / perPage);
```

Comment:
We have a Pagination component that should be able to handle this logic and UI for it. https://developers.grafana.com/ui/canary/index.html?path=/story/buttons-pagination--with-pages

Reply:
il probably do this in another PR! will have a look